src/uraft/saunafs-uraft-helper.in

```diff
@@ -151,6 +151,30 @@ saunafs_metadata_version() {
 	esac
 }
 
+is_ip_present() {
+	local ip="${1}"
+	local iface="${2:-}"
+	local -a ip_command_extra_args=()
+	if [ -n "${iface}" ]; then
+		ip_command_extra_args+=("dev" "${iface}")
+	fi
+	ip addr show "${ip_command_extra_args[@]}" | grep -wq -m1 "${ip}"
+}
+
+saunafs_is_floating_ip_present() {
+	load_config
+	if is_ip_present "${ipaddr}" "${iface}"; then
+		echo -n alive
+		return 0
+	fi
+	if [ -n "${ipaddr2}" ] && is_ip_present "${ipaddr2}" "${iface2}"; then
+		echo -n alive
+		return 0
+	fi
+	echo -n dead
```

Comment:
Consider returning different status for this case where network interface is not present. Consider scenario where network interface is gone for master service is still running, so process is still alive, and return service as `dead` could mislead administrators.

Reply:
The responsibility of this function is only to check whether floating IP is available. That's why, it only prints `alive` (floating IP is available) or `dead` (otherwise).

The responsibility about master service remains the same in `saunafs_isalive`, which returns `alive` even if floating IP is lost and `dead` when the service is stopped.